baselines/depthfl/README.md

```diff
@@ -0,0 +1,172 @@
+python -m depthfl.main fit_config.feddyn=false fit_config.kd=false # DepthFL (FedAvg)
+python -m depthfl.main fit_config.feddyn=false fit_config.kd=false fit_config.extended=false # InclusiveFL
+```
+
+To run using HeteroFL:
+```bash
+# since sbn takes too long, we test global model every 50 rounds. 
+python -m depthfl.main --config-name="heterofl" # HeteroFL
+python -m depthfl.main --config-name="heterofl" exclusive_learning=true model_size=1 # exclusive learning - 100% (a)
+```
+
+### Stateful clients comment
+
+To implement `feddyn`, stateful clients that store prev_grads information are needed. Since flwr does not yet officially support stateful clients, it was implemented as a temporary measure by loading `prev_grads` from disk when creating a client, and then storing it again on disk after learning. Specifically, there are files that store the state of each client in the `prev_grads` folder. When the strategy is instantiated (for both `FedDyn` and `HeteroFL`) the content of `prev_grads` is reset. 
+
+
+## Expected Results
+
+With the following command we run DepthFL (FedDyn / FedAvg), InclusiveFL, and HeteroFL to replicate the results of table 2,3,4 in DepthFL paper. Tables 2, 3, and 4 may contain results from the same experiment in multiple tables. 
+
+```bash
+## table 2
+python -m depthfl.main # table 2 & 4
+python -m depthfl.main exclusive_learning=true model_size=1
+python -m depthfl.main exclusive_learning=true model_size=2
+python -m depthfl.main exclusive_learning=true model_size=3
+python -m depthfl.main exclusive_learning=true model_size=4
+
+## table 2 & 3
+
+# HeteroFL & corresponding excluive learning
+python -m depthfl.main --config-name="heterofl" 
+python -m depthfl.main --config-name="heterofl" exclusive_learning=true model_size=1 model.scale=false
+python -m depthfl.main --config-name="heterofl" exclusive_learning=true model_size=2 model.scale=false
+python -m depthfl.main --config-name="heterofl" exclusive_learning=true model_size=3 model.scale=false
+python -m depthfl.main --config-name="heterofl" exclusive_learning=true model_size=4 model.scale=false
+# DepthFL (FedAvg) & corresponding exclusive learning
+python -m depthfl.main fit_config.feddyn=false fit_config.kd=false  
+python -m depthfl.main fit_config.feddyn=false fit_config.kd=false  exclusive_learning=true model_size=1
+python -m depthfl.main fit_config.feddyn=false fit_config.kd=false  exclusive_learning=true model_size=2
+python -m depthfl.main fit_config.feddyn=false fit_config.kd=false  exclusive_learning=true model_size=3
+python -m depthfl.main fit_config.feddyn=false fit_config.kd=false  exclusive_learning=true model_size=4
+
+## table 3
+python -m depthfl.main fit_config.feddyn=false fit_config.kd=false fit_config.extended=false
+
+## table 4
+python -m depthfl.main fit_config.kd=false
+python -m depthfl.main dataset_config.iid=false 
+python -m depthfl.main dataset_config.iid=false fit_config.kd=false
```

Comment:
Correct me if i'm wrong but these commands won't generate all results in Table 4 below. Should we replace with:
```
python -m depthfl.main --multirun fit_config.kd=true,false dataset_config.iid=true,false model_size=1,2,3,4
```
The above will run 16 experiments. How should we edit the command above to also generate the "ensemble" column?